Test failure: testMergeReadCancelledOnClose

**Problem description**
```
java.lang.AssertionError: Expected entry to have been cancelled upon closing
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at io.pravega.segmentstore.server.reading.ContainerReadIndexTests.testMergeReadCancelledOnClose(ContainerReadIndexTests.java:1087)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Problem location**
Segment store unit tests.

**Suggestions for an improvement**
Fix the test case.

--- a/segmentstore/server/src/test/java/io/pravega/segmentstore/server/reading/ContainerReadIndexTests.java
+++ b/segmentstore/server/src/test/java/io/pravega/segmentstore/server/reading/ContainerReadIndexTests.java
@@ -48,6 +48,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Random;
 import java.util.UUID;
+import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.TimeUnit;
@@ -1082,9 +1083,15 @@ public class ContainerReadIndexTests extends ThreadPooledTestSuite {
         // we can do is check periodically until that is done.
         TestUtils.await(entry::hasSecondEntrySet, 10, TIMEOUT.toMillis());
 
-        // Close the index and verify the entry is cancelled.
+        // Close the index.
         context.readIndex.close();
-        Assert.assertTrue("Expected entry to have been cancelled upon closing", entry.getContent().isCancelled());
+
+        // Verify the entry is cancelled. Invoke get() since the cancellation is asynchronous so it may not yet have
+        // been executed; get() will block until that happens.
+        AssertExtensions.assertThrows(
+                "Expected entry to have been cancelled upon closing",
+                () -> entry.getContent().get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS),
+                ex -> ex instanceof CancellationException);
     }
 
     /**

